Convert README from Markdown to ReStructured Text and use as longdesc

StephenWeber opened the pull request:
Discussion in #13758 led to deciding to switch README to rst and having
setup.py consume it as the long_description. Fixes #13234

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
documentation

##### ANSIBLE VERSION
```
ansible 1.4.5
```

##### SUMMARY
Permanently migrate from Markdown to ReStructured Text: github displays it properly, and it's what the PyPI expects.

StephenWeber committed:
Convert README from Markdown to ReStructured Text and use as longdesc

Discussion in #13758 led to deciding to switch README to rst and having
setup.py consume it as the long_description.

StephenWeber committed:
Fix long string in setup.py for pep8 compliance

gundalow commented:
Discussed and agreed in https://github.com/ansible/community/issues/156#issuecomment-284455925
Just need someone that knows `setup.py` to double check that.

Thanks for taking this on and fixing it

abadger commented on setup.py:
```diff
@@ -18,10 +18,21 @@
+
+
+if os.path.exists('README.rst'):
+    with open('README.rst', 'r+') as readme_file:
```
This should not be opened with "+" (as that allows writing the file).  I can't remember off the top of my head whether python2/python3 setuptools expects text strings or byte strings for longdesc.  My guess is text strings.  Since there isn't any non-ascii characted in README.rst I think we can ignore that for now and just use native strings but we might need to figure it out in the future.

StephenWeber replied:
Both are great points!

abadger commented:
I only saw one problem with setup.py.  There is one more thing that you should do.... grep -r README.md in the source tree then update various files that reference the toplevel README.md to reference the README.rst instead.  My quick grep finds a lot of other README.md's being referenced but the ones I know need to be changed are in MANIFEST.in and the ones under packaging/

StephenWeber committed:
Open README.rst as read-only

StephenWeber committed:
Update usages of root README.md to README.rst

Unsure about the file ./packaging/debian/docs as it (only) contains the
text README.md. I believe it's referencing ./packaging/debian/README.md
but maybe someone who knows debian packaging could review it?

StephenWeber commented:
I had one part of the "replace usages of README.md with README.rst" part that I wasn't 100% sure on: the file ./packaging/debian/docs as it (only) contains the text README.md. I believe it's referencing ./packaging/debian/README.md but maybe someone who knows Debian packaging could review it?

abadger committed:
Pick up fixes that had been merged into README.md after the initial conversion to rst

abadger commented:
Sorry for taking so long to get back to this.  Your changes look good.  I've rebased against current devel.

@webknjaz I know it's not as nice to do this with imperative code inside of setup.py as it is with declarative data in setup.cfg but given the older setuptools we're stuck with on the build system.  Does this look like it will work to you?

@nitzmahone We probably want this before 2.5.1 so that it will upload a long description to pypi.  We should also test it by uploading to https://test.pypi.org/ to make sure that it creates a well formatted description.  I think you were uploading test packages there while enhancing our building to pypi?  Would you want to take that action item?

webknjaz commented on README.rst:
```diff
@@ -0,0 +1,104 @@
+Read the documentation and more at https://ansible.com/
+
+You can find installation instructions
+`here <https://docs.ansible.com/intro_getting_started.html>`__ for a
```
RST requires just one trailing underscore in links

webknjaz commented on setup.py:
```diff
@@ -136,29 +136,16 @@ def run(self):
-    if filename_match:
-        with open(os.path.join(extra_requirements_dir, extra_requirements_filename)) as extra_requirements_file:
-            extra_requirements[filename_match.group(1)] = extra_requirements_file.read().splitlines()
+longdesc = 'Ansible is a radically simple IT automation system. It handles configuration-management, application deployment, cloud provisioning, ad-hoc ' \
```
I think, it's nicer to go for multiline `"""` or use parentheses for combining multiple strings instead of escaping line breaks, like:
```python
longdesc = (
    'Ansible is a radically simple IT automation system. It handles configuration-management, application deployment, cloud provisioning, ad-hoc '
    'task-execution, and multinode orchestration - including trivializing things like zero downtime rolling updates with load balancers.\n'
    '\n'
    'Read the documentation and more at https://ansible.com/'
)
```
or
```python
import textwrap
longdesc = textwrap.dedent("""
    Ansible is a radically simple IT automation system. It handles configuration-management, application deployment, cloud provisioning, ad-hoc task-execution, and multinode orchestration - including trivializing things like zero downtime rolling updates with load balancers.

    Read the documentation and more at https://ansible.com/
""")
```

webknjaz commented on setup.py:
```diff
@@ -136,29 +136,16 @@ def run(self):
+    'Read the documentation and more at https://ansible.com/'
 
 
+if os.path.exists('README.rst'):
```
I'd go for `try/except IOError` in order to make a point that variable initialized above is in fact a fallback.

abadger commented:
@webknjaz Good points, Updated the code for those and also fixed some code that I accidentally removed from the setup.py when I rebased.

webknjaz commented on setup.py:
```diff
@@ -158,6 +158,17 @@ def run(self):
+try:
+    with open('README.rst', 'r') as readme_file:
+        longdesc = readme_file.read()
+except (IOError, OSError):
```
@abadger are there any real cases where README is stripped off the distribution?

abadger replied:
Oops,  didn't se this before merging.  I don't see a reason.  that we need to check for README.rst to exist.  @StephenWeber  do you remember if there was a reason you had that in there?

StephenWeber replied:
There wasn't a reason, it probably came along from the example sources I referenced for this kind of work - there is probably some "safety first" reasoning but I can't imagine this failing (or necessarily wanting it to succeed if the file's not present).

abadger commented:
Failure this time is unrelated... failed to talk to the macos box to run tests there.  Merging to devel
